fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Multiply.java

```diff
@@ -44,7 +45,11 @@ public Expression withChildren(List<Expression> children) {
 
     @Override
     public DataType getDataType() {
-        return left().getDataType().promotion();
+        DataType rightType = child(0).getDataType();
+        DataType leftType = child(1).getDataType();
+        DataType outputType = TypeCoercionUtils.findTightestCommonType(null,
+                rightType, leftType).orElseGet(() -> rightType);
+        return outputType;
```

Comment:
maybe we need refer legacy planner to do binary arithmetic type coercion and get data type.

Reply:
we need refactor getDateType() for all functions